039-user-operator-custom-filter-for-labels-.md

```diff
@@ -0,0 +1,139 @@
+    topicOperator: {}
+    userOperator: {}
+```
+Then, we would obtain a value from the environment variable and parse it in the `UserOperatorConfig` class.
+Based on the environment variable value, we will determine what to do (a) env is not specified, and therefore we do not exclude any label from Secret
+(b) env is specified by a regex, and then we will filter and remove such labels which match the input regex.
+
+### More implementation details
+
+### 1 Parsing part
+
+As we mentioned details about the `UserOperatorConfig` class, we will elaborate more on the specific details in this section.
+Moreover, we will obtain the environment variable inside the `fromMap` method used to construct the class.
+```java
+public static UserOperatorConfig fromMap(Map<String, String> map) {
+    ...
+    // 
+    String strimziLabelsExclusionPatternEnvVar = map.get(UserOperatorConfig.STRIMZI_LABELS_EXCLUSION_PATTERN);
+    
+    if (strimziLabelsExclusionPattern != null) {
+        strimziLabelsExclusionPattern = strimziLabelsExclusionPatternEnvVar;
+    }
+    ...
+}
+```
+
+### 2 Exclusion part
+
+The exclusion part will be placed in the `KafkaUserModel` class.
+Moreover, we would need to get the value of the environment variable to such a class.
+In the `UserOperatorConfig` class, we will obtain value, and then in the `KafkaUserOperaor` class, specifically in the following method:
+```java
+protected Future<KafkaUserStatus> createOrUpdate(Reconciliation reconciliation, KafkaUser resource) {
+    KafkaUserModel user;
+    KafkaUserStatus userStatus = new KafkaUserStatus();
+
+    try {
+        user = KafkaUserModel.fromCrd(resource, config.getSecretPrefix(), config.isAclsAdminApiSupported(),config.isKraftEnabled(), 
+            config.getStrimziLabelsExclusionPattern); // <-- this one we will inject into KafkaUserModel class
+    } catch (Exception e) {
+        StatusUtils.setStatusConditionAndObservedGeneration(resource, userStatus, Future.failedFuture(e));
+        return Future.failedFuture(new ReconciliationException(userStatus, e));
+    }
+    ...
+}
+```
+Then, we can use such value to store it as an instance attribute of `KafkaUserModel` and implement post-processing of labels in the
+following method:
+```java
+protected Secret createSecret(Map<String, String> data) {
+    final Secret secret = return new SecretBuilder()
+        .withNewMetadata()
+            .withName(getSecretName())
+            .withNamespace(namespace)
+            .withLabels(Util.mergeLabelsOrAnnotations(labels.toMap(), templateSecretLabels))
+            .withAnnotations(Util.mergeLabelsOrAnnotations(null, templateSecretAnnotations))
+            .withOwnerReferences(createOwnerReference())
+        .endMetadata()
+        .withType("Opaque")
+        .withData(data)
+        .build();
+    
+    // here, we have to do post-processing (i.e., filtering) of labels by exclusion pattern
+    // filter by value of instance variable `this.getStrimziLabelsExclusionPattern`
```

Comment:
Well, it could be done here or before the creation of an instance. But I think pre-processing would be better in this case. I have changed the order :)